tests/EStimLibrary.UnitTests/Extensions/SpatialModel/StringHierarchy/StringHierarchyLocationFactoryTests.cs

```diff
@@ -0,0 +1,299 @@
+﻿using EStimLibrary.Core.SpatialModel;
+using EStimLibrary.Extensions.SpatialModel.StringHierarchy;
+
+
+namespace EStimLibrary.UnitTests.Extensions.SpatialModel.StringHierarchy;
+
+
+/// <summary>
+/// Test class for StringHierarchyLocationFactory.
+/// </summary>
+public class StringHierarchyLocationFactoryTests
+{
+    private readonly ITestOutputHelper _output;
+
+    // Test class constructor creates an output helper so can write console
+    // output.
+    public StringHierarchyLocationFactoryTests(
+        ITestOutputHelper testOutputHelper)
+    {
+        this._output = testOutputHelper;
+    }
+
+    // Test method naming convention: LibClassMethodName_ScenarioShouldExpectn
+
+    /// <summary>
+    /// Test the constructor with null parameter.
+    /// </summary>
+    [Fact]
+    public void Constructor_Null_ShouldError()
+    {
+        bool caughtError = false;
+        try
+        {
+            // Create factory with null base region
+            StringHierarchyLocationFactory factory =
+                new StringHierarchyLocationFactory(null!);
+        }
+        catch (NullReferenceException) { caughtError = true; }
+
+        // Check that error occurred
+        Assert.True(caughtError);
+    }
+
+    /// <summary>
+    /// Test the constructor with non-null parameter.
+    /// </summary>
+    [Fact]
+    public void Constructor_NonNull_ShouldInit()
```

Comment:
If it doesn't make the test method name too long, try keeping to this <MethodName>_<Scenario>_Should<ExpectedBehavior/Output> format